contracts/src/chains/OPStackKeystore.sol

```diff
@@ -45,6 +25,45 @@ abstract contract OPStackKeystore is Keystore {
     /// @notice Thrown when the provided OutputRoot preimages do not has to the expected OutputRoot.
     error InvalidL2OutputRootPreimages();
 
+    ////////////////////////////////////////////////////////////////////////////////////////////////////
+    //                                            STRUCTURES                                          //
+    ////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    /// @dev Proof used to extract a Keystore config hash from an OPStack master L2.
+    struct OPStackProof {
+        /// @dev The L1 state root proof.
+        L1StateRootLib.L1StateRootProof l1StateRootProof;
+        /// @dev The `AnchorStateRegistry` storage proof on L1.
+        StorageProof anchorStateRegistryProof;
+        /// @dev The Keystore storage proof on the master L2.
+        StorageProof masterKeystoreProof;
+        /// @dev The preimages of the OutputRoot.
+        PartialOutputRootPreimages partialOutputRootPreimages;
+        /// @dev The master L2 block header, encoded in RLP format.
+        ///      NOTE: Its hash is computed to complete the `partialOutputRootPreimages`.
+        bytes masterL2BlockHeaderRlp;
```

Comment:
It might be worth moving this into the `OutputRootPreimages` struct so it's not "partial" anymore and doesn't need so much explanation. I think it's easier to explain that the header needs to be hashed first.